libs/ledger/src/chain/block_coordinator.cpp

```diff
@@ -586,65 +548,15 @@ BlockCoordinator::State BlockCoordinator::OnPreExecBlockValidation()
       return fail("No previous block in chain");
     }
 
-    // Check that the weight as given by the proof is correct
-    if (stake_)
+    if (consensus_)
     {
-      // Attempt to ascertain the beacon value within the block
-      uint64_t random_beacon = 0;
-
-      // Try to get entropy for the block we are generating - it must eventually resolve to OK or
-      // FAILED
-      for (uint16_t i = 0;;)
-      {
-        auto result = beacon_->GenerateEntropy(current_block_->body.hash,
-                                               current_block_->body.block_number, random_beacon);
-
-        if (result == EntropyGeneratorInterface::Status::OK)
-        {
-          break;
-        }
-
-        if (result == EntropyGeneratorInterface::Status::NOT_READY)
-        {
-          FETCH_LOG_INFO(LOGGING_NAME,
-                         "Waiting for entropy for block: ", current_block_->body.block_number);
-        }
-
-        if (result == EntropyGeneratorInterface::Status::FAILED ||
-            (result == EntropyGeneratorInterface::Status::OK &&
-             random_beacon != current_block_->body.random_beacon))
-        {
-          return fail("Failed to verify entropy during block validation. Discarding block.");
-        }
-
-        std::this_thread::sleep_for(std::chrono::milliseconds(5));
-
-        if (i++ == 5)
-        {
-          state_machine_->Delay(std::chrono::milliseconds{100});
-          return fail("Timed out waiting for entropy when validating block", false);
-        }
-      }
-
-      if (current_block_->body.random_beacon != random_beacon)
-      {
-        FETCH_LOG_INFO(LOGGING_NAME, "Saw incorrect random beacon from: ",
-                       current_block_->body.miner.address().ToBase64(),
-                       ".Block number: ", current_block_->body.block_number,
-                       " expected: ", random_beacon, " got: ", current_block_->body.random_beacon);
-        return fail("Block has incorrect random beacon ");
-      }
-
-      if (!stake_->ValidMinerForBlock(*previous, current_block_->body.miner))
-      {
-        return fail("Block signed by miner deemed invalid by the staking mechanism");
-      }
+      consensus_->UpdateCurrentBlock(*previous);  // Only update with valid blocks
+      auto result = consensus_->ValidBlock(*previous, *current_block_);
 
-      if (current_block_->weight !=
-              stake_->GetBlockGenerationWeight(*previous, current_block_->body.miner) &&
-          current_block_->weight != 0)
+      if (!(result == ConsensusInterface::Status::YES ||
```

Comment:
Reasonably confusing does it make sense to simple say == NO

Reply:
Not to my mind - this is explicitly continuting even in the presence of 'unknown' (this will go away if we put the group sig in the block however)